test/blueprint/keyframe.spec.ts

```diff
@@ -0,0 +1,510 @@
+											// Katapult k8s adapter does not kow how to handle it (as a opposed to sw.containerized-{something}),
+											// so it falls back to declaring a secret reference according to the capabilities defined by sw.database.
+											valueFrom: {
+												secretKeyRef: {
+													name: 'api-config',
+													key:
+														'integration.postgres-db.default-endpoint.connection-uri',
+												},
+											},
+										},
+										{
+											name: 'integration.redis.default-endpoint.connection-uri',
+											valueFrom: {
+												secretKeyRef: {
+													name: 'api-config',
+													key:
+														'integration.redis.default-endpoint.connection-uri',
+												},
+											},
+										},
+									],
+								},
+							],
+						},
+					},
+				},
+			});
+
+			// Check jellyfish-ui deployment.
+			expect(k8sSpecs.deployments[1]).to.deep.equal({
+				apiVersion: 'app/v1',
+				kind: 'Deployment',
+
+				metadata: {
+					namespace: 'productOs',
+					name: 'ui',
+				},
+
+				spec: {
+					replicas: 4,
+
+					selector: {
+						matchLabels: {
+							'app.kubernetes.io/name': 'ui',
+						},
+					},
+
+					template: {
+						metadata: {
+							labels: {
+								'app.kubernetes.io/name': 'api',
+							},
+						},
+
+						spec: {
+							containers: [
+								{
+									name: 'ui',
+									imagePullPolicy: 'always',
+									image: 'balena/jellyfish-ui:vTODO',
+
+									// jellyfish-ui contract has type sw.containerized-service which provides an endpoint, which defines
+									// what goes below.
+									ports: [
+										{
+											name: 'default-endpoint',
+											containerPort: 80,
+											protocol: 'TCP',
+										},
+									],
+									readinessProbe: {
+										httpGet: {
+											path: '/health',
+											port: 'default-endpoint',
+											scheme: 'http',
+										},
+									},
+									// TODO: Add livenessProbe.
+
+									env: [
+										// From requirements.
+										{
+											// Name pattern: integration.{name from "contract.requires"}.{expand what is defined by the type of the dependency}
+											name:
+												'integration.jellyfish-api.api-endpoint.connection-uri',
+											// jellyfish-api dependency of ui is handled by connections in the blueprinnt.
+											// katapult is responsible for declaring a service for jeellyfish-api and sets the value according to
+											// the service name.
+											value: 'http://api',
+										},
+									],
+								},
+							],
+						},
+					},
+				},
+			});
+		});
+
+		it('hav correct service definitions', () => {
+			// Check jellyfish-api service.
+			expect(k8sSpecs.services[0]).to.deep.equal({
+				apiVersion: 'app/v1',
+				kind: 'Service',
+
+				metadata: {
+					namespace: 'productOs',
+					name: 'api',
+				},
+
+				spec: {
+					type: 'LoadBalancer', // TODO: Where does this come from?
```

Comment:
@ab77 Do we have a decided approach how we expose services? Should we define services backed with AWS loadbalancers or ingress?

I see that JF is currently using ingress, and balenaCloud seems to be leveraging LB-backed services.